client/src/components/StudentMain.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from "react";
+import QuestionForm from "./QuestionForm.js";
+import { fetchDialogs, fetchMessages, createNewDialog, postQuestion } from "../services/api.js";
+import translateServerErrors from "../services/translateServerErrors.js";
+import ErrorList from "./layout/ErrorList.js";
+
+const StudentMain = (props) => {
+    const dialogId = props.match.params.id
+    const [ dialogs, setDialogs ] = useState([])
+    const [ questions, setQuestions ] = useState([])
+    const [ noDialogMessage, setNoDialogMessage ] = useState("")
+    const [ errors, setErrors ] = useState([])
+
+    const showQuestions = questions.map(question => {
+        return (
+            <li className="question" key={question.id}>
+                {question.content}
+            </li>
+        )
+    })
+
+    const showDialogs = dialogs.map(dialogListId => {
+        return (
+            <li className="dialog" key={dialogListId}>
+                <a href={`/ask/${dialogListId}`}>{dialogListId}</a>
```

Comment:
if this is linking to a page in your application, we should use the react `Link` element which helps to avoid a full page reload from a new HTTP request, as the Link instead makes use of the BrowserRouter